src/engines/roundRobin.ts

```diff
@@ -4,19 +4,69 @@ import { LiteEngine } from "./engine";
 export class LiteRoundRobinEngine implements LiteEngine {
 
     readonly engines: LiteEngine[];
+    
+    #closed = false;
+    #counter = 0;
 
     constructor(engines: LiteEngine[]) {
         this.engines = engines;
     }
 
-    query<REQ, RES>(f: TLFunction<REQ, RES>, req: REQ, args: { timeout: number, awaitSeqno?: number }): Promise<RES> {
-        let id = Math.floor(Math.random() * this.engines.length);
-        return this.engines[id].query(f, req, args);
+    async query<REQ, RES>(f: TLFunction<REQ, RES>, req: REQ, args: { timeout: number, awaitSeqno?: number }): Promise<RES> {
+        if(this.#closed) {
+            throw new Error('Engine is closed');
+        }
+
+        let attempts = 0
+        let id = this.#counter++ % this.engines.length
+        let errorsCount = 0
```

Comment:
Can you, please, rewrite to **backoff** from **teslabot**?

Reply:
I don't think backoff is apropriate here, because we're not querying same engine, we iterate over them, so we want to do that as fast as possible. In most cases counter iteration occurs when timeout occurs.